safehttp/plugins/fetch_metadata/fetch_metadata.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fetchmetadata
+
+import (
+	"github.com/google/go-safeweb/safehttp"
+)
+
+const (
+	enforceMode = "enforce"
+	reportMode  = "report"
+)
+
+// LoggingService TODO
+type LoggingService interface {
+	Log(ir *safehttp.IncomingRequest)
+}
+
+// Plugin implements Fetch Metadata functionality.
+// See https://www.w3.org/TR/fetch-metadata/ for more details.
+type Plugin struct {
+	mode   string
```

Comment:
Change `mode` to a bool called `ReportOnly`. This will be false by default. That is, enforcement is on by default.

Or another solution: Remove `mode` and just check whether logger is nil or not to check which mode is enabled.

Reply:
Right, so you could have only the `logger` and remove `reportOnly`. And if `logger` is `nil` then enforcement is on and if it isn't `nil` then it's reporting mode.

Reply:
I'd rather keep explicit functions to switch between report and enforce mode with explanations on what the user is about to do. :)